SecurityException: Permission Denial: startForeground requires android.permission.FOREGROUND_SERVICE

Maybe a missing permission?
With the latest master, this crash happens anytime I try to upload an image (via either gallery or camera, from either home or nearby).
Happens even after uninstalling and reinstalling the app.
I ACRA'd a bunch yesterday. Logcat:
```
10-28 20:43:04.837  6022  6022 D AndroidRuntime: Shutting down VM
10-28 20:43:04.838  6022  6022 E AndroidRuntime: FATAL EXCEPTION: main
10-28 20:43:04.838  6022  6022 E AndroidRuntime: Process: fr.free.nrw.commons, PID: 6022
10-28 20:43:04.838  6022  6022 E AndroidRuntime: java.lang.RuntimeException: Unable to start service fr.free.nrw.commons.upload.UploadService@2b26a5 with Intent { act=fr.free.nrw.commons.upload.upload cmp=fr.free.nrw.commons/.upload.UploadService }: java.lang.SecurityException: Permission Denial: startForeground from pid=6022, uid=10380 requires android.permission.FOREGROUND_SERVICE
10-28 20:43:04.838  6022  6022 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3903)
10-28 20:43:04.838  6022  6022 E AndroidRuntime: 	at android.app.ActivityThread.access$1700(ActivityThread.java:236)
10-28 20:43:04.838  6022  6022 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1815)
10-28 20:43:04.838  6022  6022 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
10-28 20:43:04.838  6022  6022 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:214)
10-28 20:43:04.838  6022  6022 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7032)
10-28 20:43:04.838  6022  6022 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
10-28 20:43:04.838  6022  6022 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
10-28 20:43:04.838  6022  6022 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:965)
10-28 20:43:04.838  6022  6022 E AndroidRuntime: Caused by: java.lang.SecurityException: Permission Denial: startForeground from pid=6022, uid=10380 requires android.permission.FOREGROUND_SERVICE
10-28 20:43:04.838  6022  6022 E AndroidRuntime: 	at android.os.Parcel.createException(Parcel.java:1966)
10-28 20:43:04.838  6022  6022 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:1934)
10-28 20:43:04.838  6022  6022 E AndroidRuntime: 	at android.os.Parcel.readException(Parcel.java:1884)
10-28 20:43:04.838  6022  6022 E AndroidRuntime: 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:5043)
10-28 20:43:04.838  6022  6022 E AndroidRuntime: 	at android.app.Service.startForeground(Service.java:695)
10-28 20:43:04.838  6022  6022 E AndroidRuntime: 	at fr.free.nrw.commons.upload.UploadService.onStartCommand(UploadService.java:227)
10-28 20:43:04.838  6022  6022 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3884)
10-28 20:43:04.838  6022  6022 E AndroidRuntime: 	... 8 more
10-28 20:43:04.838  6022  6022 E AndroidRuntime: Caused by: android.os.RemoteException: Remote stack trace:
10-28 20:43:04.838  6022  6022 E AndroidRuntime: 	at com.android.server.am.ActivityManagerService.enforcePermission(ActivityManagerService.java:12150)
10-28 20:43:04.838  6022  6022 E AndroidRuntime: 	at com.android.server.am.ActiveServices.setServiceForegroundInnerLocked(ActiveServices.java:1289)
10-28 20:43:04.838  6022  6022 E AndroidRuntime: 	at com.android.server.am.ActiveServices.setServiceForegroundLocked(ActiveServices.java:969)
10-28 20:43:04.838  6022  6022 E AndroidRuntime: 	at com.android.server.am.ActivityManagerService.setServiceForeground(ActivityManagerService.java:24824)
10-28 20:43:04.838  6022  6022 E AndroidRuntime: 	at android.app.IActivityManager$Stub.onTransact$setServiceForeground$(IActivityManager.java:11378)
10-28 20:43:04.838  6022  6022 E AndroidRuntime: 
10-28 20:43:04.838  6022  6022 E ACRA    : ACRA caught a RuntimeException for fr.free.nrw.commons
10-28 20:43:04.838  6022  6022 E ACRA    : java.lang.RuntimeException: Unable to start service fr.free.nrw.commons.upload.UploadService@2b26a5 with Intent { act=fr.free.nrw.commons.upload.upload cmp=fr.free.nrw.commons/.upload.UploadService }: java.lang.SecurityException: Permission Denial: startForeground from pid=6022, uid=10380 requires android.permission.FOREGROUND_SERVICE
10-28 20:43:04.838  6022  6022 E ACRA    : 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3903)
10-28 20:43:04.838  6022  6022 E ACRA    : 	at android.app.ActivityThread.access$1700(ActivityThread.java:236)
10-28 20:43:04.838  6022  6022 E ACRA    : 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1815)
10-28 20:43:04.838  6022  6022 E ACRA    : 	at android.os.Handler.dispatchMessage(Handler.java:106)
10-28 20:43:04.838  6022  6022 E ACRA    : 	at android.os.Looper.loop(Looper.java:214)
10-28 20:43:04.838  6022  6022 E ACRA    : 	at android.app.ActivityThread.main(ActivityThread.java:7032)
10-28 20:43:04.838  6022  6022 E ACRA    : 	at java.lang.reflect.Method.invoke(Native Method)
10-28 20:43:04.838  6022  6022 E ACRA    : 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
10-28 20:43:04.838  6022  6022 E ACRA    : 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:965)
10-28 20:43:04.838  6022  6022 E ACRA    : Caused by: java.lang.SecurityException: Permission Denial: startForeground from pid=6022, uid=10380 requires android.permission.FOREGROUND_SERVICE
10-28 20:43:04.838  6022  6022 E ACRA    : 	at android.os.Parcel.createException(Parcel.java:1966)
10-28 20:43:04.838  6022  6022 E ACRA    : 	at android.os.Parcel.readException(Parcel.java:1934)
10-28 20:43:04.838  6022  6022 E ACRA    : 	at android.os.Parcel.readException(Parcel.java:1884)
10-28 20:43:04.838  6022  6022 E ACRA    : 	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:5043)
10-28 20:43:04.838  6022  6022 E ACRA    : 	at android.app.Service.startForeground(Service.java:695)
10-28 20:43:04.838  6022  6022 E ACRA    : 	at fr.free.nrw.commons.upload.UploadService.onStartCommand(UploadService.java:227)
10-28 20:43:04.838  6022  6022 E ACRA    : 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3884)
10-28 20:43:04.838  6022  6022 E ACRA    : 	... 8 more
10-28 20:43:04.838  6022  6022 E ACRA    : Caused by: android.os.RemoteException: Remote stack trace:
10-28 20:43:04.838  6022  6022 E ACRA    : 	at com.android.server.am.ActivityManagerService.enforcePermission(ActivityManagerService.java:12150)
10-28 20:43:04.838  6022  6022 E ACRA    : 	at com.android.server.am.ActiveServices.setServiceForegroundInnerLocked(ActiveServices.java:1289)
10-28 20:43:04.838  6022  6022 E ACRA    : 	at com.android.server.am.ActiveServices.setServiceForegroundLocked(ActiveServices.java:969)
10-28 20:43:04.838  6022  6022 E ACRA    : 	at com.android.server.am.ActivityManagerService.setServiceForeground(ActivityManagerService.java:24824)
10-28 20:43:04.838  6022  6022 E ACRA    : 	at android.app.IActivityManager$Stub.onTransact$setServiceForeground$(IActivityManager.java:11378)
10-28 20:43:04.838  6022  6022 E ACRA    : 
10-28 20:43:04.869  6022  6022 I Process : Sending signal. PID: 6022 SIG: 9
10-28 20:43:04.970  5120  5120 I Zygote  : Process 6022 exited due to signal (9)
10-28 20:43:04.971  5613  5660 I libprocessgroup: Successfully killed process cgroup uid 10380 pid 6022 in 5ms
```

--- a/app/src/main/AndroidManifest.xml
+++ b/app/src/main/AndroidManifest.xml
@@ -17,6 +17,7 @@
     <uses-permission android:name="android.permission.MANAGE_ACCOUNTS" />
     <uses-permission android:name="com.google.android.apps.photos.permission.GOOGLE_PHOTOS" />
     <uses-permission android:name="android.permission.SET_WALLPAPER"/>
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
 
 
     <!-- Needed only if your app targets Android 5.0 (API level 21) or higher. -->

